src/modules/peek/Peek.FilePreviewer/FilePreview.xaml.cs

```diff
@@ -93,7 +94,11 @@ private async Task OnFilePropertyChanged()
                 return;
             }
 
-            Previewer = previewerFactory.Create(File);
+            // Cancel previous loading task
+            _cancellationTokenSource.Cancel();
+            _cancellationTokenSource = new ();
+
+            Previewer = previewerFactory.Create(File, _cancellationTokenSource.Token);
             await UpdatePreviewAsync();
```

Comment:
I think it might be good to add a  ThrowIfCancelled before the async calls in UpdatePreviewAsync. I'm not sure if it's necessary - depends on if this exception can crash the app/ui thread, but a corresponding try and catch block in On FilePropertyChanged could be added for this

Reply:
Added, and also added the try catch.